src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -195,11 +195,29 @@ static void rr_subchannel_list_unref(grpc_exec_ctx *exec_ctx,
 static void rr_subchannel_list_shutdown(grpc_exec_ctx *exec_ctx,
                                         rr_subchannel_list *subchannel_list,
                                         const char *reason) {
+  if (subchannel_list->shutting_down) {
+    if (GRPC_TRACER_ON(grpc_lb_round_robin_trace)) {
+      gpr_log(GPR_DEBUG, "Subchannel list %p already shutting down",
+              (void *)subchannel_list);
+    }
+    return;
```

Comment:
Yes, and they are very common. Consider: four `sd`s that share/belong to a given subchannel_list. When this subchannel list is outdated, it'll be shutdown (line 556). But because the connectivity callback works at the `sd` level, this will happen for all `sd`s that share/belong to that subchannel list. See the following log for an example:
```
round_robin.c:527] [RR 0x7fa87d6d0e80] connectivity changed for sl 0x7fa87d55c9c0, subchannel 0x7fa87d5d0d40: prev_state=1 new_state=1 p->shutdown=0 sd->subchannel_list->shutting_down=0 error="No Error"
round_robin.c:557] about to shutdown OUTDATED Subchannel list 0x7fa87d55c9c0
round_robin.c:205] Shutting down subchannel_list 0x7fa87d55c9c0
round_robin.c:214] Unsubscribing from subchannel 0x7fa87d76a3e0 as part of shutting down subchannel_list 0x7fa87d55c9c0
round_robin.c:214] Unsubscribing from subchannel 0x7fa87d5d0d40 as part of shutting down subchannel_list 0x7fa87d55c9c0
round_robin.c:214] Unsubscribing from subchannel 0x7fa87d5d03e0 as part of shutting down subchannel_list 0x7fa87d55c9c0
round_robin.c:214] Unsubscribing from subchannel 0x7fa87d5d0b60 as part of shutting down subchannel_list 0x7fa87d55c9c0
round_robin.c:182] [RR 0x7fa87d6d0e80] subchannel_list 0x7fa87d55c9c0 UNREF 5->4
round_robin.c:527] [RR 0x7fa87d6d0e80] connectivity changed for sl 0x7fa87d55c9c0, subchannel 0x7fa87d76a3e0: prev_state=1 new_state=1 p->shutdown=0 sd->subchannel_list->shutting_down=1 error="No Error"
round_robin.c:557] about to shutdown OUTDATED Subchannel list 0x7fa87d55c9c0
round_robin.c:198] Subchannel list 0x7fa87d55c9c0 already shutting down
round_robin.c:527] [RR 0x7fa87d6d0e80] connectivity changed for sl 0x7fa87d55c9c0, subchannel 0x7fa87d5d03e0: prev_state=1 new_state=1 p->shutdown=0 sd->subchannel_list->shutting_down=1 error="No Error"
round_robin.c:557] about to shutdown OUTDATED Subchannel list 0x7fa87d55c9c0
round_robin.c:198] Subchannel list 0x7fa87d55c9c0 already shutting down
round_robin.c:527] [RR 0x7fa87d6d0e80] connectivity changed for sl 0x7fa87d55c9c0, subchannel 0x7fa87d5d0b60: prev_state=1 new_state=1 p->shutdown=0 sd->subchannel_list->shutting_down=1 error="No Error"
round_robin.c:557] about to shutdown OUTDATED Subchannel list 0x7fa87d55c9c0
round_robin.c:198] Subchannel list 0x7fa87d55c9c0 already shutting down
```